### PR TITLE
Refactor of disk initialization

### PR DESCRIPTION
**Pull request by netcho**
We're employing the new tool called `disk_init_helper` to discover drives and format them. This new tool requires a new set of parameters, so we've updated the inventory file.

### CODE REVIEW

File: hosts.example [hunk 1]
```diff
@@ -105,58 +101,80 @@
...
+[storpool_servers:vars]
+# Drive selection section
+#   - Values can be obtained by running 'lsblk -do NAME,SIZE,MODEL' on the nodes
+#   - Devices matching _any_ of the conditions below will get initialized
+#   - Larger NVME drives will be split into equally-sized partitions smaller than 4T
+#   - A matched device, already containing partitions, will trigger an error
+#     remove partitions manually or override with sp_drive_erase=True
 #
+# CAUTION: The drive_init procedure is not fully implemented yet
+#   - No RAID controller detection, and private case scenarios
```
Comment (krokodilerian) [inline hosts.example:132]: what's "private case"?

File: hosts.example [hunk 2]
```diff
@@ -105,58 +101,80 @@
...
+
+# Minimal size of PMEM drives (SSD and HDD) below which drives won't be discovered to be used by StorPool
+# sp_min_pmem_size='1GB'
+
+# Whether to run the discover phase of disk_init_helper. Set to false to use a pre-generated JSON file to be
+# used for disk initialization
+# sp_discover_disk_run=True
+
+# Disk glob pattern to discover disks by. Specify manufacturer model name, serial number.
+# sp_discover_disk_pattern=''
```
Comment (krokodilerian) [inline hosts.example:151]: is there a default, or by default it doesn't match anything?

also, an example would be great

File: roles/initialize_disks/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+Role Name
```
Comment (krokodilerian) [inline roles/initialize_disks/README.md:1]: Should we not leave a "lorem ipsum" file, at least we can add  "TBD" or something

File: roles/initialize_disks/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,112 @@
+---
+- name: Retrieving node ID from storpool.conf
+  command: storpool_showconf -n -e SP_OURID
+  register: retrieve_id_result
+  tags:
+    - check_configuration
+
+- name: Checking if host is defined in storpool.conf
```
Comment (krokodilerian) [inline roles/initialize_disks/tasks/main.yml:8]: this check is not just for disk roles

File: roles/initialize_disks/meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
+galaxy_info:
+  author: Kaloyan Kotlarski
+  description: your role description
+  company: StorPool Inc.
```
Comment (krokodilerian) [inline roles/initialize_disks/meta/main.yml:4]: This is wrong, it's StorPool AD, and we should fix the rest of this file, too (are we using apache or bsd license for this, for example?)

File: roles/initialize_disks/files/perform_disktest.sh [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+	if [ -f "/etc/storpool/${serial}.tested" ]; then
+		continue
+	fi
+	echo `basename "$i"` >> drives.txt
+	echo "$serial" >> serials.txt
+done
+
+if [ -f drives.txt ]; then
+	for d in `cat drives.txt`; do
+		parted -s /dev/${d} mklabel msdos
```
Comment (krokodilerian) [inline roles/initialize_disks/files/perform_disktest.sh:59]: this is probably just a copy, but this is very very dangerous, the whole script needs to be checked

**Review by krokodilerian — CHANGES_REQUESTED** [review]

File: hosts.example (hunk 1 above)
Comment (netcho) [reply hosts.example:132]: This was a left-over from the old section

File: hosts.example (hunk 2 above)
Comment (netcho) [reply hosts.example:151]: I've added a link to the user guide that shows how pattern matching works

File: roles/initialize_disks/README.md (hunk 3 above)
Comment (netcho) [reply roles/initialize_disks/README.md]: I've updated it to reflect better the purpose of the role

File: roles/initialize_disks/tasks/main.yml (hunk 4 above)
Comment (netcho) [reply roles/initialize_disks/tasks/main.yml:8]: I moved it back to the stage where requirements are checked

File: roles/initialize_disks/meta/main.yml (hunk 5 above)
Comment (netcho) [reply roles/initialize_disks/meta/main.yml:4]: Updated

File: roles/initialize_disks/files/perform_disktest.sh (hunk 6 above)
Comment (krokodilerian) [reply roles/initialize_disks/files/perform_disktest.sh:59]: please comment out this line

**Review by krokodilerian — CHANGES_REQUESTED** [review]

**Pull request merged**